File: internal/service/events.go
package service

// Event names (D00 CONTRACTS §3, closed enum). No other event name may be
// emitted anywhere in internal/service (B02 SPEC §2.11). TS union mirror:
// packages/core/src/events.ts.
const (
	EventConfigChanged   = "config:changed"
	EventCatalogChanged  = "catalog:changed"
	EventUsageUpdated    = "usage:updated"
	EventSettingsChanged = "settings:changed"
	EventPickRecorded    = "pick:recorded"
)

File: packages/core/src/events.ts
import type { GUISettings } from './types.js'

// Closed event enum (D00 CONTRACTS §3).
export const ENGINE_EVENTS = [
  'config:changed',
  'catalog:changed',
  'usage:updated',
  'settings:changed',
  'pick:recorded',
] as const

export type EngineEvent = (typeof ENGINE_EVENTS)[number]

export type EngineEventPayloads = {
  'config:changed': { section: string }
  'catalog:changed': Record<string, never>
  'usage:updated': Record<string, never>
  'settings:changed': GUISettings
  'pick:recorded': { profile_slug: string; route_key: string }
}
